src/react/elements.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { Realm } from "../realm.js";
+import { AbstractValue, AbstractObjectValue, Value, ObjectValue, FunctionValue, NullValue } from "../values/index.js";
+import { Create, Properties } from "../singletons.js";
+import { TypesDomain, ValuesDomain } from "../domains/index.js";
+import invariant from "../invariant.js";
+import { Get } from "../methods/index.js";
+import { getReactSymbol, objectHasNoPartialKeyAndRef, deleteRefAndKeyFromProps } from "./utils.js";
+import * as t from "babel-types";
+import { computeBinary } from "../evaluators/BinaryExpression.js";
+import { CompilerDiagnostic, FatalError } from "../errors.js";
+
+function createPropsObject(
+  realm: Realm,
+  type: Value,
+  config: ObjectValue | AbstractValue | AbstractObjectValue | NullValue,
+  children: Value
+) {
+  let defaultProps = type instanceof ObjectValue ? Get(realm, type, "defaultProps") : realm.intrinsics.undefined;
+  let props = Create.ObjectCreate(realm, realm.intrinsics.ObjectPrototype);
+  // start by having key and ref deleted, if they actually exist, they will be add later
+  deleteRefAndKeyFromProps(realm, props);
+  let key = realm.intrinsics.null;
+  let ref = realm.intrinsics.null;
+
+  const setProp = (name: string, value: Value): void => {
+    if (name === "children") {
```

Comment:
What is special about this case? I don't see how it differs from the last catch-all.